src/ConnectionPool.cpp

```diff
@@ -446,8 +446,14 @@ err_code_t ConnectionPool::waitPoll() {
 
         if (pollfd_ptr->revents & (POLLERR | POLLHUP | POLLNVAL)) {
           markDeadConn(conn, keywords::kCONN_POLL_ERROR, pollfd_ptr);
-          ret_code = RET_CONN_POLL_ERR;
-          --m_nActiveConn;
+          if (conn->tryReconnect()) {
```

Comment:
Use a macro to avoid duplicate codes?

Reply:
I don't like to use macro here, how about create a function `ConnectionPoll::rewind`? just like `markDeadConn`.

Reply:
lgtm